jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/internal/HttpReceiverOverHTTP.java

```diff
@@ -294,7 +295,6 @@ else if (read == 0)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("Error processing {} in {}", endPoint, this, x);
-            releaseNetworkBuffer();
```

Comment:
This certainly deserves a comment that explains `close()` is going to take care of releasing the buffer but with a thread protection thanks to a CAS.